### PR TITLE
Add pointers to higher-order time-stepping schemes

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
Adds pointers to Niklas' and Tobias' work on solving the heat equation using implicit Runge Kutta methods and pySDC.

This is a draft, because we have to wait until #557 is merged and Tobias' thesis is published on MediaTUM.

Checklist:

- [x] I added a summary of any user-facing changes (compared to the last release) in the `changelog-entries/<PRnumber>.md`.
- [x] I will remember to squash-and-merge, providing a useful summary of the changes of this PR.
- [x] Add pointers in source code to Tobias' thesis.

### CODE REVIEW

**Comment by BenjaminRodenberg** [comment]
@MakisH is there a better place to put this information? We could also add a pointer to #415 and #557 but I prioritized providing pointers to the thesis documents. The PRs are visible in the history. The link to the thesis documents is not so clear.

File: partitioned-heat-conduction/README.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ preCICE configuration (image generated using the [precice-config-visualizer](htt
 
 You can either couple a solver with itself or different solvers with each other. In any case you will need to have preCICE and the python bindings installed on your system.
 
-* FEniCS. Install [FEniCS](https://fenicsproject.org/download/) and the [FEniCS-adapter](https://github.com/precice/fenics-adapter). The code is largely based on this [fenics-tutorial](https://github.com/hplgit/fenics-tutorial/blob/master/pub/python/vol1/ft03_heat.py) from [1].
+* FEniCS. Install [FEniCS](https://fenicsproject.org/download/) and the [FEniCS-adapter](https://github.com/precice/fenics-adapter). The code is largely based on this [fenics-tutorial](https://github.com/hplgit/fenics-tutorial/blob/master/pub/python/vol1/ft03_heat.py) from [1]. In `fenics-solver`, You can find alternative version of the solver that offering higher-order time-stepping schemes: `heatHigherOrder.py` offers higher-order implicit Runge Kutta schemes [4] and `heat_pySDC.py` spectral deferred correction methods and pySDC [5].
```
Comment (MakisH) [inline partitioned-heat-conduction/README.md]:
```suggestion
* FEniCS. Install [FEniCS](https://fenicsproject.org/download/) and the [FEniCS-adapter](https://github.com/precice/fenics-adapter). The code is largely based on this [fenics-tutorial](https://github.com/hplgit/fenics-tutorial/blob/master/pub/python/vol1/ft03_heat.py) from [1]. In `fenics-solver`, You can find alternative versions of the solver, offering higher-order time-stepping schemes: `heatHigherOrder.py` offers higher-order implicit Runge Kutta schemes [4], and `heat_pySDC.py` offers spectral deferred correction methods and pySDC [5].
```

File: partitioned-heat-conduction/README.md [hunk 2]
```diff
@@ -87,3 +87,5 @@ Visualization in paraview for `x_c = 1.5`.
 [1] Hans Petter Langtangen and Anders Logg. "Solving PDEs in Minutes-The FEniCS Tutorial Volume I." (2016). [pdf](https://fenicsproject.org/pub/tutorial/pdf/fenics-tutorial-vol1.pdf)  
 [2] Azahar Monge and Philipp Birken. "Convergence Analysis of the Dirichlet-Neumann Iteration for Finite Element Discretizations." (2016). Proceedings in Applied Mathematics and Mechanics. [doi](https://doi.org/10.1002/pamm.201610355)  
 [3] Benjamin Rüth, Benjamin Uekermann, Miriam Mehl, Philipp Birken, Azahar Monge, and Hans Joachim Bungartz. "Quasi-Newton waveform iteration for partitioned surface-coupled multiphysics applications." (2020). International Journal for Numerical Methods in Engineering. [doi](https://doi.org/10.1002/nme.6443)  
+[4] Niklas Vinnitchenko. "Evaluation of Higher-Order Coupling Schemes with FEniCS-preCICE." (2024). Bachelor's thesis at Technical University of Munich. [url](https://mediatum.ub.tum.de/1732367)  
+[5] Thesis Tobias Eppacher, "TODO" (2024). Bachelor's thesis at Technical University of Munich. [url](TODO)
```
Comment (MakisH) [inline partitioned-heat-conduction/README.md]: TODO

File: partitioned-heat-conduction/README.md [hunk 3]
```diff
@@ -87,3 +87,5 @@ Visualization in paraview for `x_c = 1.5`.
 [1] Hans Petter Langtangen and Anders Logg. "Solving PDEs in Minutes-The FEniCS Tutorial Volume I." (2016). [pdf](https://fenicsproject.org/pub/tutorial/pdf/fenics-tutorial-vol1.pdf)  
 [2] Azahar Monge and Philipp Birken. "Convergence Analysis of the Dirichlet-Neumann Iteration for Finite Element Discretizations." (2016). Proceedings in Applied Mathematics and Mechanics. [doi](https://doi.org/10.1002/pamm.201610355)  
 [3] Benjamin Rüth, Benjamin Uekermann, Miriam Mehl, Philipp Birken, Azahar Monge, and Hans Joachim Bungartz. "Quasi-Newton waveform iteration for partitioned surface-coupled multiphysics applications." (2020). International Journal for Numerical Methods in Engineering. [doi](https://doi.org/10.1002/nme.6443)  
+[4] Niklas Vinnitchenko. "Evaluation of Higher-Order Coupling Schemes with FEniCS-preCICE." (2024). Bachelor's thesis at Technical University of Munich. [url](https://mediatum.ub.tum.de/1732367)
```
Comment (MakisH) [inline partitioned-heat-conduction/README.md]:
```suggestion
[4] Niklas Vinnitchenko. "Evaluation of Higher-Order Coupling Schemes with FEniCS-preCICE." (2024). Bachelor's thesis at Technical University of Munich. [online](https://mediatum.ub.tum.de/1732367)  
```

**Review by MakisH — APPROVED** [review]
I think this is the perfect place for this information.

File: partitioned-heat-conduction/README.md [hunk 4]
```diff
@@ -31,7 +31,7 @@ preCICE configuration (image generated using the [precice-config-visualizer](htt
 
 You can either couple a solver with itself or different solvers with each other. In any case you will need to have preCICE and the python bindings installed on your system.
 
-* FEniCS. Install [FEniCS](https://fenicsproject.org/download/) and the [FEniCS-adapter](https://github.com/precice/fenics-adapter). The code is largely based on this [fenics-tutorial](https://github.com/hplgit/fenics-tutorial/blob/master/pub/python/vol1/ft03_heat.py) from [1].
+* FEniCS. Install [FEniCS](https://fenicsproject.org/download/) and the [FEniCS-adapter](https://github.com/precice/fenics-adapter). The code is largely based on this [fenics-tutorial](https://github.com/hplgit/fenics-tutorial/blob/master/pub/python/vol1/ft03_heat.py) from [1]. In `fenics-solver`, You can find alternative versions of the solver, offering higher-order time-stepping schemes: `heatHigherOrder.py` offers higher-order implicit Runge Kutta schemes [4], and `heat_pySDC.py` offers spectral deferred correction methods and pySDC [5].
```
Comment (BenjaminRodenberg) [inline partitioned-heat-conduction/README.md]:
```suggestion
* FEniCS. Install [FEniCS](https://fenicsproject.org/download/) and the [FEniCS-adapter](https://github.com/precice/fenics-adapter). The code is largely based on this [fenics-tutorial](https://github.com/hplgit/fenics-tutorial/blob/master/pub/python/vol1/ft03_heat.py) from [1]. In `fenics-solver`, You can find alternative versions of the solver, offering higher-order time-stepping schemes: `heatHigherOrder.py` offers higher-order implicit Runge Kutta schemes [4], and `heat_pySDC.py` offers spectral deferred correction methods and pySDC [5]. (Hint: use `./run.sh -t highorder` or `./run.sh -t sdc`)
```
Too much @MakisH ?

Comment (MakisH) [reply partitioned-heat-conduction/README.md]: No, this is a useful hint.

File: partitioned-heat-conduction/README.md [hunk 5]
```diff
@@ -87,3 +87,5 @@ Visualization in paraview for `x_c = 1.5`.
 [1] Hans Petter Langtangen and Anders Logg. "Solving PDEs in Minutes-The FEniCS Tutorial Volume I." (2016). [pdf](https://fenicsproject.org/pub/tutorial/pdf/fenics-tutorial-vol1.pdf)  
 [2] Azahar Monge and Philipp Birken. "Convergence Analysis of the Dirichlet-Neumann Iteration for Finite Element Discretizations." (2016). Proceedings in Applied Mathematics and Mechanics. [doi](https://doi.org/10.1002/pamm.201610355)  
 [3] Benjamin Rüth, Benjamin Uekermann, Miriam Mehl, Philipp Birken, Azahar Monge, and Hans Joachim Bungartz. "Quasi-Newton waveform iteration for partitioned surface-coupled multiphysics applications." (2020). International Journal for Numerical Methods in Engineering. [doi](https://doi.org/10.1002/nme.6443)  
+[4] Niklas Vinnitchenko. "Evaluation of Higher-Order Coupling Schemes with FEniCS-preCICE." (2024). Bachelor's thesis at Technical University of Munich. [online](https://mediatum.ub.tum.de/1732367)  
+[5] Thesis Tobias Eppacher, "TODO" (2024). Bachelor's thesis at Technical University of Munich. [url](TODO)
```
Comment (BenjaminRodenberg) [inline partitioned-heat-conduction/README.md]:
```suggestion
[5] Thesis Tobias Eppacher, "Parallel-in-Time Integration with preCICE" (2024). Bachelor's thesis at Technical University of Munich. [url](https://mediatum.ub.tum.de/1755012)
```

**Comment by BenjaminRodenberg** [comment]
Closing this PR. Superseeded by #574. Added 0ccab4e04e350f48928fca5f41422ed4f4e2225f directly to `develop`.

**Pull request closed without merging**